### ISSUE TITLE
allow injection of additional CFLAGS

### ISSUE THREAD

**Issue by umlaeute** [rbdannenberg/o2#21]
it seems that the CMakeFile sets the build flags https://github.com/rbdannenberg/o2/blob/151b1c6b9536fac333e647a13a6cef3b547fa814/CMakeLists.txt#L361 in a way, that does not allow the user to inject additional build flags.

would it be possible to use something like this instead:

```cmake
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c11 -mcx16")
```

?

**Comment by rbdannenberg** [comment]
Yes, I made the change. As far as I can tell, with the change, you can set CMAKE_C_FLAGS in the cache to "extra" flags and get "-std=c11 -mcx16" added automatically.

**Comment by umlaeute** [comment]
> As far as I can tell, [..]

yes, this was my intention.

> Yes, I made the change

i don't see it in current master: https://github.com/rbdannenberg/o2/blob/e83488f166bb5d58ba672217dd0c2a3cf3031874/CMakeLists.txt#L362

am i missing something?
